### PR TITLE
File validation fails when file contains sub-arrays

### PR DESCRIPTION
**Pull request by jonhudson**

### CODE REVIEW

**Comment by weierophinney** [comment]
Please update the pull request summary to describe what this accomplishes; since I link to the PRs in the changelog, that's often the first place people look to understand how the change impacts them.

To be clear: we already support sub-arrays, but specifically _indexed_ sub-arrays, vs _associative_ sub-arrays, which is what this PR addresses.

File: src/ServerRequestFactory.php [hunk 1]
```diff
@@ -469,6 +469,7 @@ private static function createUploadedFileFromSpec(array $value)
      */
     private static function normalizeNestedFileSpec(array $files = [])
     {
+        $normalizedFiles = [];
```
Comment (weierophinney) [inline src/ServerRequestFactory.php]: There's no need to create a new array here.

Comment (weierophinney) [inline src/ServerRequestFactory.php]: I stand corrected when running the unit tests. Still trying to understand why this change results in any change in behavior…

**Comment by weierophinney** [comment]
I've updated the summary for you.

File: src/ServerRequestFactory.php (hunk 1 above)
Comment (jonhudson) [inline src/ServerRequestFactory.php]: The `UploadedFile` objects were being appended to the `$files` array, which meant that the `tmp_name` and other attributes preceeded them in the array. When same array gets passed to the validation function, it was trying to then validate strings such as the `tmp_name` as an instance of `UploadedFile`, and failing. So I just created a new array to store the `UploadedFile` objects in and returned that instead.

**Comment by jonhudson** [comment]
Thank you.
